### PR TITLE
[ios] Refactored for new search API.

### PR DESCRIPTION
**Pull request by igrechuhin**

### CODE REVIEW

File: iphone/Maps/Classes/Search/MWMSearch.mm [hunk 1]
```diff
@@ -50,56 +52,77 @@ - (instancetype)initManager
...
+    if (results.IsEndMarker())
+    {
+      self.everywhereSearchActive = NO;
+      [self onSearchCompleted];
+    }
+    else
+    {
+      self->m_results = results;
+      self.suggestionsCount = results.GetSuggestsCount();
+      self.resultsCount = results.GetCount();
```
Comment (vmihaylenko) [inline iphone/Maps/Classes/Search/MWMSearch.mm]:
``` objc
- (NSUInteger)resultsCount { return results.GetCount(); }
```

File: iphone/Maps/Classes/Search/MWMSearch.mm [hunk 2]
```diff
@@ -137,10 +160,12 @@ + (void)searchQuery:(NSString *)query forInputLocale:(NSString *)inputLocale
   if (inputLocale && inputLocale.length != 0)
```
Comment (vmihaylenko) [inline iphone/Maps/Classes/Search/MWMSearch.mm]: тут не нужно оба условия, достаточно 2го

File: iphone/Maps/Classes/Search/MWMSearch.mm (hunk 1 above)
Comment (igrechuhin) [inline iphone/Maps/Classes/Search/MWMSearch.mm]: OK

File: iphone/Maps/Classes/Search/MWMSearch.mm (hunk 2 above)
Comment (igrechuhin) [inline iphone/Maps/Classes/Search/MWMSearch.mm]: OK

**Comment by vmihaylenko** [comment]
LGTM

File: iphone/Maps/Classes/Search/MWMSearch.mm [hunk 3]
```diff
@@ -50,56 +51,76 @@ - (instancetype)initManager
...
+    if (!self)
+      return;
+    if (results.IsEndMarker())
+    {
+      self.everywhereSearchActive = NO;
+      [self onSearchCompleted];
+    }
+    else
+    {
+      self->m_results = results;
```
Comment (ygorshenin) [inline iphone/Maps/Classes/Search/MWMSearch.mm]: onResults callback may be called multiple times. Is it OK that results is assigned, not appended to m_results?

Comment (ygorshenin) [inline iphone/Maps/Classes/Search/MWMSearch.mm]: OK, seems good.

**Comment by ygorshenin** [comment]
LGTM

**Pull request merged**